### PR TITLE
Hide the published blog posts on the tag detail if the publish date hasn't passed yet

### PR DESCRIPTION
**Pull request by carakas**
## Type

- Non critical bugfix

## Pull request description
published blog post that where scheduled to become visible in the future where already visible on tag detail
if you clicked on it you got a 404
This pr prevents them from showing up on the tag detail

### CODE REVIEW

File: src/Frontend/Modules/Blog/Engine/Model.php [hunk 1]
```diff
@@ -580,9 +580,9 @@ public static function getForTags(array $ids)
             'SELECT i.title, m.url
              FROM blog_posts AS i
              INNER JOIN meta AS m ON m.id = i.meta_id
-             WHERE i.status = ? AND i.hidden = ? AND i.id IN (' . implode(',', $ids) . ')
+             WHERE i.status = ? AND i.hidden = ? AND i.id IN (' . implode(',', $ids) . ') AND i.publish_on <= ?
              ORDER BY i.publish_on DESC',
-            array('active', 'N')
+            array('active', 'N', FrontendModel::getUTCDate('Y-m-d H:i') . ':00')
```
Comment (StijnVrolijk) [inline src/Frontend/Modules/Blog/Engine/Model.php]: Why append the seconds manually? Does SQL have improved performance if all times are rounded off to a minute?

File: src/Frontend/Modules/Blog/Engine/Model.php [hunk 2]
```diff
@@ -719,7 +719,7 @@ public static function getRecentComments($limit = 5)
              WHERE c.status = ? AND i.status = ? AND i.language = ? AND i.hidden = ? AND i.publish_on <= ?
              ORDER BY c.id DESC
              LIMIT ?',
-            array('published', 'active', LANGUAGE, 'N', FrontendModel::getUTCDate('Y-m-d H:i') . ':00', $limit)
+            array('published', 'active', LANGUAGE, 'N', FrontendModel::getUTCDate('Y-m-d H:i'), $limit)
```
Comment (StijnVrolijk) [inline src/Frontend/Modules/Blog/Engine/Model.php]: Same here, why no seconds?

Comment (carakas) [inline src/Frontend/Modules/Blog/Engine/Model.php]: I removed them everywhere in 9690090

File: src/Frontend/Modules/Blog/Engine/Model.php [hunk 3]
```diff
@@ -47,7 +47,7 @@ public static function get($URL)
              INNER JOIN meta AS m2 ON c.meta_id = m2.id
              WHERE i.status = ? AND i.language = ? AND i.hidden = ? AND i.publish_on <= ? AND m.url = ?
              LIMIT 1',
-            array('active', LANGUAGE, 'N', FrontendModel::getUTCDate('Y-m-d H:i') . ':00', (string) $URL)
+            array('active', LANGUAGE, 'N', FrontendModel::getUTCDate('Y-m-d H:i'), (string) $URL)
```
Comment (StijnVrolijk) [inline src/Frontend/Modules/Blog/Engine/Model.php]: Hold it right there, sonny. Will comparing `Y-m-d H:i` with `Y-m-d H:i:s` work>

Comment (carakas) [inline src/Frontend/Modules/Blog/Engine/Model.php]: it does, just tested it

Comment (StijnVrolijk) [inline src/Frontend/Modules/Blog/Engine/Model.php]: very well, just fix the merge conflicts and you are on your way to becoming a fork master!

**Comment by carakas** [comment]
fixed the merge conflicts

**Pull request merged**